content/v2.0/query-data/flux/calculate-percentages.md

```diff
@@ -0,0 +1,247 @@
+| 2020-01-01T00:00:50Z | gpu          | mem_total | 8589934592 |
+
+### Pivot fields into columns
+Use `pivot()` to pivot the `mem_used` and `mem_total` fields into columns.
+Output includes `mem_used` and `mem_total` columns with values for each corresponding `_time`.
+
+```js
+// ...
+  |> pivot(rowKey:["_time"], columnKey: ["_field"], valueColumn: "_value")
+```
+
+###### Returns the following:
+
+| _time                | _measurement | mem_used   | mem_total  |
+|:-----                |:------------:| --------:  | ---------: |
+| 2020-01-01T00:00:00Z | gpu          | 2517924577 | 8589934592 |
+| 2020-01-01T00:00:10Z | gpu          | 2695091978 | 8589934592 |
+| 2020-01-01T00:00:20Z | gpu          | 2576980377 | 8589934592 |
+| 2020-01-01T00:00:30Z | gpu          | 3006477107 | 8589934592 |
+| 2020-01-01T00:00:40Z | gpu          | 3543348019 | 8589934592 |
+| 2020-01-01T00:00:50Z | gpu          | 4402341478 | 8589934592 |
+
+### Map new values
+With fields pivoted into columns, each row contains the values necessary to calculate a percentage.
+Use `map()` to re-map values in each row.
+Divide `mem_used` by `mem_total` and multiply by 100 to return the percentage.
+
+{{% note %}}
+To return a precise float percentage value that includes decimal points the example
+below casts integer field values to floats and multiplies by a float value (`100.0`).
+{{% /note %}}
+
+```js
+// ...
+  |> map(fn: (r) => ({
+    _time: r._time,
+    _measurement: r._measurement,
+    _field: "mem_used_percent",
+    _value: float(v: r.mem_used) / float(v: r.mem_total) * 100.0
+  }))
+```
+
+| _time                | _measurement | _field           | _value  |
+|:-----                |:------------:|:------:          | ------: |
+| 2020-01-01T00:00:00Z | gpu          | mem_used_percent | 29.31   |
+| 2020-01-01T00:00:10Z | gpu          | mem_used_percent | 31.37   |
+| 2020-01-01T00:00:20Z | gpu          | mem_used_percent | 30.00   |
+| 2020-01-01T00:00:30Z | gpu          | mem_used_percent | 35.00   |
+| 2020-01-01T00:00:40Z | gpu          | mem_used_percent | 41.25   |
+| 2020-01-01T00:00:50Z | gpu          | mem_used_percent | 51.25   |
+
+### Full query
+```js
+from(bucket: "gpu-monitor")
+  |> range(start: 2020-01-01T00:00:00Z)
+  |> filter(fn: (r) => r._measurement == "gpu" and r._field =~ /mem_/ )
+  |> pivot(rowKey:["_time"], columnKey: ["_field"], valueColumn: "_value")
+  |> map(fn: (r) => ({
+    _time: r._time,
+    _measurement: r._measurement,
+    _field: "mem_used_percent",
+    _value: float(v: r.mem_used) / float(v: r.mem_total) * 100.0
+  }))
+```
+
+---
+
+## Pivot vs join
+To use values in mathematical operations in Flux, operand values must exists in a single row.
+Both `pivot()` and `join()` will do this, but there are important differences between the two:
+
+#### Pivot is more performant
+`pivot()` only has to read and operate on a single stream of data.
+`join()` requires two streams of data and the overhead of reading and combing both
```

Comment:
join() reads and operates on two streams of data, requiring more overhead, especially